tests/fileTrashTest.js

```diff
@@ -177,18 +463,135 @@ describe('oc.fileTrash', function () {
     })
 
     describe('and when this deleted folder is restored to a different location', function () {
-      let deletedFolderId
-      let originalLocation
-      beforeEach(function (done) {
-        if (!trashEnabled) {
-          pending()
-        }
-        originalLocation = testFolder + ' (restored to a different location)'
-        oc.fileTrash.list('/').then(trashItems => {
-          deletedFolderId = trashItems[1].getName()
-          done()
-        })
+      const originalLocation = testFolder + ' (restored to a different location)'
+
+      beforeAll(function (done) {
+        const promises = []
+        promises.push(setGeneralInteractions(provider))
+        promises.push(provider.addInteraction({
+          uponReceiving: 'MOVE folder from trashbin to fileslist to a different location',
+          withRequest: {
+            method: 'MOVE',
+            path: Pact.Matchers.term({
+              matcher: '.*\\/remote\\.php\\/dav\\/trash-bin\\/admin\\/' + config.deletedFolderId,
+              generate: '/remote.php/dav/trash-bin/admin/' + config.deletedFolderId
+            }),
+            headers: {
+              Destination: config.owncloudURL + 'remote.php/dav/files/admin/' + config.testFolder + '%20(restored%20to%20a%20different%20location)'
+            }
+          },
+          willRespondWith: {
+            status: 201,
+            headers: {
+              'Content-Type': 'text/html; charset=utf-8',
+              'Access-Control-Allow-Origin': origin,
+              'Access-Control-Allow-Headers': accessControlAllowHeaders,
+              'Access-Control-Allow-Method': accessControlAllowMethods
+            }
+          }
+        }))
+        promises.push(provider.addInteraction({
+          uponReceiving: 'PROPFIND to an empty trashbin after restoring',
```

Comment:
looks same like `'PROPFIND to an empty trashbin',` can we use one interaction for both?

Reply:
Yes, we can use one interaction for both. Need to do some refactoring for that which is WIP.